src/store/filestore.rs

```diff
@@ -49,206 +43,196 @@ impl Store for LocalFileStore {
     }
 
     fn delete_annotation<T>(&self, name: &str, version: &str) -> Result<()> {
-        // Search the name ver index for the hash
-        let hash = self.get_hash_from_name_ver_tree::<T>(name, version)?;
+        let hash = self.lookup_hash::<T>(name, version)?;
+        let (count, _) = Self::parse_annotation_path(
+            &self.make_path::<T>(&hash, Self::make_annotation_relpath("*", "*")),
+        )?
+        .size_hint();
+        if count == 1 {
+            return Err(OrcaError::from(Kind::DeletingLastAnnotation(
+                get_type_name::<T>(),
+                name.to_owned(),
+                version.to_owned(),
+            )));
+        }
+        let annotation_file =
+            self.make_path::<T>(&hash, &Self::make_annotation_relpath(name, version));
+        fs::remove_file(&annotation_file)?;
 
-        fs::remove_file(self.make_annotation_path::<T>(&hash, name, version))?;
         Ok(())
     }
 }
 
 impl LocalFileStore {
-    /// New function that takes the directory as where to save the files
+    /// Construct a local file store instance in a specific directory.
     pub fn new(directory: impl AsRef<Path>) -> Self {
         Self {
             directory: directory.as_ref().into(),
         }
     }
-
-    /// Getter function for directory
+    /// Get the directory where store is located.
     pub fn get_directory(&self) -> &Path {
         &self.directory
     }
-
-    fn make_dir_path<T>(&self, hash: &str) -> PathBuf {
+    /// Relative path where model specification is stored within the model directory.
+    pub const SPEC_RELPATH: &str = "spec.yaml";
+    /// Relative path where model annotation is stored within the model directory.
+    pub fn make_annotation_relpath(name: &str, version: &str) -> PathBuf {
+        PathBuf::from(format!("annotation/{name}-{version}.yaml"))
+    }
+    /// Build the storage path with the model directory (`hash`) and a file's relative path.
+    pub fn make_path<T>(&self, hash: &str, relpath: impl AsRef<Path>) -> PathBuf {
         PathBuf::from(format!(
             "{}/{}/{}",
             self.directory.to_string_lossy(),
             get_type_name::<T>(),
-            hash,
+            hash
         ))
+        .join(relpath)
+    }
+
+    fn parse_annotation_path(path: &Path) -> Result<impl Iterator<Item = Result<ModelInfo>>> {
```

Comment:
Perhaps instead of path, do glob search pattern because that more accurately describes what it is?